components/image/picker/src/index.js

```diff
@@ -0,0 +1,134 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+
+import ImagePlaceholder from '@schibstedspain/sui-image-placeholder'
+
+class ImagePicker extends Component {
+  constructor (props) {
+    super(props)
+
+    const { images } = props
+    let filteredImages = []
+    let selected = -1
+
+    if (images.length > 0) {
+      filteredImages = images.filter(image => image.src && image.thumb)
+
+      if (filteredImages.length > 0) {
+        selected = filteredImages.findIndex(({ selected }) => selected)
+        selected = (selected === -1) ? 0 : selected
+      }
+    }
+
+    this.state = {
+      selected,
+      images: filteredImages
+    }
+  }
+
+  _getHeadImage () {
+    const { images, selected } = this.state
+    const { emptyImage } = this.props
+
+    const placeHolder = {
+      alt: 'Placeholder',
+      src: emptyImage
+    }
+
+    const src = (images.length === 0) ? emptyImage : images[selected].src
+    const alt = (images.length === 0) ? 'No text available' : images[selected].alt
+
+    return (<ImagePlaceholder src={src} alt={alt} placeholder={placeHolder} />)
+  }
+
+  _getListImages () {
+    const { images, selected } = this.state
+
+    const imageList = images.map((image, index) => {
+      const itemClass = cx('sui-ImagePicker-thumbItem', {
+        'sui-ImagePicker-thumbItem--selected': index === selected
+      })
+      return (
+        <li className={itemClass} key={image.src} onClick={() => this._handleClick(image)}>
+          <img src={image.thumb} alt={image.alt} title={image.title} />
+        </li>
+      )
+    })
+    return imageList
+  }
+
+  _handleClick (clickedImage) {
+    const { images } = this.state
+    const { onClick } = this.props
+
+    const selected = images.findIndex(image => image.src === clickedImage.src)
+    this.setState({ selected })
+    onClick(clickedImage)
+  }
+
+  _buttonAction = (callback) => {
+    const { images, selected } = this.state
+    if (images.length === 0 || selected === -1) { return }
```

Comment:
According to the code above, if there's no images, `selected` if define as -1 anyways. So, the first condition is unnecessary.